docs/custom-domain/README.md

```diff
@@ -1,56 +1,96 @@
-
-### Deploy and configure an Nginx Docker image on a new Azure VM
-
-Deploy via Azure Portal
+The following steps describe how you can achieve this.
 
-<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Facr%2Fmaster%2Fdocs%2Fcustom-domain%2Fdocker-vm-deploy%2Fazuredeploy.json" target="_blank">
-    <img src="http://azuredeploy.net/deploybutton.png"/>
-</a>
-<a href="http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Facr%2Fmaster%2Fdocs%2Fcustom-domain%2Fdocker-vm-deploy%2Fazuredeploy.json" target="_blank">
-    <img src="http://armviz.io/visualizebutton.png"/>
-</a>
-
-Alternatively, to deploy using powershell script, [docker-vm-deploy/](docker-vm-deploy/), do the following:
-
-1. Edit [azuredeploy.parameters.json](docker-vm-deploy/azuredeploy.parameters.json) and populate all necessary parameters
-
-2. Run the following script to create the new VM:
-
-        `.\deploy.ps1 -resourceGroupName <resourceGroup>`
-
-### Configure DNS zone
-
-Configure the DNS zone so `registry.contoso.com` points to the Azure VM you have just created. If you are using an Azure DNS Zone. You can use the following command:
-
-  `New-AzureRmDnsRecordSet -Name <registry> -RecordType CNAME -ZoneName <contoso.com> -ResourceGroupName <resourceGroup> -Ttl <Ttl> -DnsRecords (New-AzureRmDnsRecordConfig -Cname <AddrToAboveVM>)`
-
-## Quick verification
-
-A simple way to test the setup is to call `docker login` to quickly confirm that the requests are properly forwarded:
-
-  `docker login -u <username> -p <password> registry.contoso.com`
+## Prerequisites
+- [Azure CLI](https://docs.microsoft.com/cli/azure/?view=azure-cli-latest): version 2.4.0 or higher
+  - Consider using [Azure Cloud Shell](https://docs.microsoft.com/azure/cloud-shell/overview)
+- A _premium_ Azure Container Registry. See [here](https://docs.microsoft.com/azure/container-registry/container-registry-get-started-azure-cli) for instructions on how to create one.
+- Your custom domain names. The following two are required:
+  - Custom registry domain to access the registry REST API. Example for the `contoso.com` domain: `container-registry.contoso.com` 
+  - Custom data domain to access the registry content. Again, example for `contoso.com`: `eastus-registry-data.contoso.com`
+    - Note that the custom data domain is region specific. For geo-replicated registries, each region should have its own custom data endpoint.
+
+  For each domain, you must prepare a single PEM formatted file containing the TLS private key and public certificate:
+  
+  ```
+  -----BEGIN PRIVATE KEY-----  
+  .....  
+  -----END PRIVATE KEY-----  
+  -----BEGIN CERTIFICATE-----  
+  .....  
+  -----END CERTIFICATE-----
+  ```
+  
+  For example, using [openssl](https://github.com/openssl/openssl):
+  - Create a self-signed public cert and private key
```

Comment:
>private key [](start = 41, length = 11)

Just highlight these are bash command on the top somehwere. 
Also break up it up as a multi-line command with \ etc.

Reply:
Marked the code as `shell` and split up the lines.